packages/grid/x-grid/src/tests/columnHeaders.DataGrid.test.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import {
+  createClientRenderStrictMode,
+  // @ts-expect-error need to migrate helpers to TypeScript
+  fireEvent,
+} from 'test/utils';
+import { expect } from 'chai';
+import { XGrid } from '@material-ui/x-grid';
+import { getColumnHeaderCell, raf } from 'test/utils/helperFn';
+
+const isJSDOM = /jsdom/.test(window.navigator.userAgent);
+
+describe('<XGrid /> - Column Headers', () => {
+  // TODO v5: replace with createClientRender
+  const render = createClientRenderStrictMode();
+
+  const baselineProps = {
+    autoHeight: isJSDOM,
+    disableColumnResize: false,
+    rows: [
+      {
+        id: 0,
+        brand: 'Nike',
+        foundationYear: 1964,
+      },
+      {
+        id: 1,
+        brand: 'Adidas',
+        foundationYear: 1949,
+      },
+      {
+        id: 2,
+        brand: 'Puma',
+        foundationYear: 1948,
+      },
+    ],
+  };
+
+  describe('GridColumnHeaderMenu', () => {
+    it('should close menu when resizing a column', async () => {
+      render(
+        <div style={{ width: 300, height: 500 }}>
+          <XGrid
+            {...baselineProps}
+            columns={[
+              { field: 'brand', resizable: true },
+              { field: 'foundationYear', resizable: true },
+            ]}
+          />
+        </div>,
+      );
+
+      expect(document.querySelectorAll('.MuiGridMenu-root')).to.have.length(0);
+
+      const menuIcon = getColumnHeaderCell(0).querySelector('.MuiDataGrid-menuIcon');
+
+      fireEvent.click(menuIcon!.querySelector('.MuiDataGrid-menuIconButton'));
+
+      expect(menuIcon).to.have.class('MuiDataGrid-menuOpen');
+
+      const separator = getColumnHeaderCell(0).querySelector('.MuiDataGrid-iconSeparator');
+      fireEvent.mouseDown(separator);
+      await raf();
```

Comment:
Do we need the raf and mouse up events to reproduce the bug in the test case?

Reply:
If I do not do the `mouseUp` event, then the next bug is failing
But maybe there is just a cleanup that needs to be done between the two tests.

Reply:
https://app.circleci.com/pipelines/github/mui-org/material-ui-x/5588/workflows/9535d3d3-6d3b-45e7-9aff-0581fab8113a/jobs/28784?invite=true#step-111-229

Reply:
To give a more recent example :

Here is the CI with the RAF / mouseUp : https://app.circleci.com/pipelines/github/mui-org/material-ui-x/5630/workflows/73b90c24-27b9-4a06-949c-00477018a3e8

Here is the CI without the RAF / mouseUp : https://app.circleci.com/pipelines/github/mui-org/material-ui-x/5632/workflows/65bf1f80-87bf-4341-a45b-4515eeec074d

With only this commit between the two : https://github.com/mui-org/material-ui-x/pull/1989/commits/034d088c4835cddf11e13c1cb9ce936c2f5d7113

I tried with just the mouseUp and not the Raf and the tests are passing if you prefer.

Reply:
```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
```

There is probably a missing cleanup somewhere when a parent of this icon is unmounted while being mousedown

Reply:
https://github.com/mui-org/material-ui-x/pull/1989/commits/351dc207be5931e8aa2e774943c87c3540669d38

This should fix this error

Reply:
Ahhh ok, it's a regression from #1720. It also explains why column resizing is broken since 4.0.0-alpha.31.